src/modules/testimonials/testimonials.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { CreateTestimonialDto } from './dto/create-testimonial.dto';
+import { Testimonial } from './entities/testimonials.entity';
+
+@Injectable()
+export class TestimonialsService {
+  constructor(
+    @InjectRepository(Testimonial)
+    private readonly testimonialRepository: Repository<Testimonial>
+  ) {}
+  async createTestimonial(createTestimonialDto: CreateTestimonialDto, user) {
```

Comment:
add a try catch block here to handle exceptions that might occur

Reply:
When you implement the try…catch in the controller it overrides whatever error your throwing from the service. @alphacodez . I used to have it but removed it because all errors become 500

Reply:
Done in the service file